### ISSUE TITLE
Question: best way to create "vhost"-like hosting

### ISSUE THREAD

**Issue by lexoyo** [silexlabs/unifile#147]
The question is: should I use the new extension system for what I am thinking about

I want to create a service, where my user authenticate and have an ID. They should be able to access a folder on my FTP or amazon storage - list, upload, download files with cloud explorer. Then I will provide them with a subdomain, e.g. {id}.mydomain.com and I want to serve their files on this subdomain.

For the subdomain part I it is easy but for the "shared" FTP I would like to use the one provided by unifile, so maybe the extension system could add the ID to all paths? E.g. "/userfolder1/file.jpg" would be "/my/ftp/root/{ID}/userfolder1/file.jpg"

**Comment by JbIPS** [comment]
Yes, I'm thinking of adding the path into the plugin lifecycle so it could be modified automatically.

From your example, you would setup a plugin to convert any path to your absolute one with ID
